src/workerd/jsg/async-context.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "jsg.h"
+#include <v8.h>
+
+namespace workerd::jsg {
+
+// Provides for basic internal async context tracking. Eventually, it is expected that
+// this will be provided by V8 assuming that the AsyncContext proposal advances through
+// TC-39. For now, however, we implement a model that is very similar to that implemented
+// by Node.js.
+
+struct AsyncResource {
+  const uint64_t id;
+  const kj::Maybe<uint64_t> parentId;
+  kj::Maybe<Value> store;
```

Comment:
Note that `store` here is not quite correct here and will be evolved. Each ALS instance must maintain its own storage context such that they do not interfere with each other. This was written this way initially to just provide something quick for the initial experiments.